operator/images/cluster-setup/bin/install.sh

```diff
@@ -108,6 +118,68 @@ switch_cluster() {
   fi
 }
 
+fetch_bitwarden_secrets() {
+  CREDENTIALS_DIR="$WORKSPACE_DIR/credentials"
+  BITWARDEN_CRED="$CREDENTIALS_DIR/secrets/bitwarden.yaml"
+
+  ( setx_off BW_CLIENTID="${BW_CLIENTID:-}" )
+  ( setx_off BW_CLIENTSECRET="${BW_CLIENTSECRET:-}" )
+  ( setx_off BW_PASSWORD="${BW_PASSWORD:-}" )
+
+  if [ ! -e "$BITWARDEN_CRED" ]; then
+    return
+  fi
+
+  printf "[Bitwarden]:\n "
+  printf "bitwarden.yaml file exists. Checking if the required variables to connect to Bitwarden are set.\n" | indent 2
+  setx_off
+  if [ -z "$BW_PASSWORD" ]; then
+      printf "Please set the required env variables and try again.\n" >&2 | indent 2
+      return
+  fi
+  setx_on
+
+  printf "Required variables are available.\n" | indent 2
+  if [ "$(bw logout >/dev/null 2>&1)$?" -eq 0 ]; then
+    printf "Logout successful.\n" >/dev/null
+  fi
+  setx_off
+  if [ "$(BW_CLIENTID="$BW_CLIENTID" BW_CLIENTSECRET="$BW_CLIENTSECRET" bw login --apikey >/dev/null 2>&1)$?" -eq 0 ]; then
+    printf "Login successful.\n" >/dev/null
+  fi
+  setx_on
+
+  login_status=$(bw login --check 2>&1)
+  if [ "$login_status" = "You are not logged in." ]; then
+    printf "Error while logging into Bitwarden.\n" >&2 | indent 2
+    return
+  fi
+
+  setx_off
+  session=$(BW_PASSWORD="$BW_PASSWORD" bw unlock --passwordenv BW_PASSWORD --raw)
+  setx_on
+
+  # process id/path pairs from bitwarden.yaml
+  secret_count=$(yq '.credentials | length' "$BITWARDEN_CRED")
+  for i in $(seq 0 "$((secret_count-1))")
+  do
+    setx_off
+    content=$(bw get password "$(yq ".credentials[$i].id" "$BITWARDEN_CRED")" --session "$session")
+    path=$(yq ".credentials[$i].path" "$BITWARDEN_CRED")
+
+    if ! mkdir -p "$(dirname "$WORKSPACE_DIR/$path")" 2>/dev/null; then
+      printf "Unable to create the folder. Exiting.\n" >&2 | indent 2
+      exit 1
+    fi
+    if ! echo "$content" | base64 -d > "$WORKSPACE_DIR/$path"; then
+      printf "Unable to copy the contents of the secret to the specified path. Exiting.\n" >&2 | indent 2
+      exit 1
+    fi
+    setx_on
```

Comment:
Add a printf statement list the file and id.

Reply:
Done.